### ISSUE TITLE
Calculating progress in a list of progress (front-end or back-end)

### ISSUE THREAD

**Issue by pablorosalesps** [surveyjs/survey-library#3213]
Hey! We currently show a list of surveys to the user, and one of the columns on that list/table shows the progress for that specific survey. We would like to calculate that progress in the backend and send it with the list of surveys. 

Is there anything on the SurveyJS ecosystem that could help us with this without us having to replicate a lot of the SurveyJS code? If not and the task is too complicated, we would love some alternative ideas. We are evaluating calculating that on-the-fly on the list itself using SurveyJS. Any suggestion would be great.

Appreciate any feedback, we already have some licenses and are loving the product.

### Are you requesting a feature, reporting a bug, or asking a question?
asking question

### What is the current behavior?
We store a set of surveys and would like to show a list of surveys with the progress (in %) on that list.

### What is the expected behavior?
If possible, calculate the % of answered questions (progress, **taking into account visible questions**) in the backend.

### Specify your

* browser:
* browser version:
* surveyjs platform (angular or react or jquery or knockout or vue):
* surveyjs version:

**Comment by andrewtelnov** [comment]
@pablorosalesps What is your server platform? It will be easy to do for NodeJS. You can create a Survey model, set survey.data (response). After that you can go though all questions `survey.getAllQuestions(true)` or visible questions `survey.getAllQuestions(true)` and check if they have value, `question.isEmpty()` function should return `false`.

Thank you,
Andrew

**Comment by pablorosalesps** [comment]
Yep, forgot to mention it was `NodeJS`. Great news. We need to make the progress based on `visible` questions only, would that complicate things?

**Comment by pablorosalesps** [comment]
Nvm, re-read your comment, you already mention the `true` parameter. We'll give that a try, thanks a lot!
